components/raftstore/src/store/fsm/peer.rs

```diff
@@ -4418,6 +4421,11 @@ where
                 }
             }
         }
+        if let Some(last_snapshot) = &self.fsm.peer.last_snapshot_time {
+            if *last_snapshot >= down_peer_limit {
```

Comment:
If the snapshot is applied already, the force compact can't be performed still. I don't think it's better than the previous strategy which is based on the `peer_heartbeats` and snapshot state

Reply:
But `peer_heartbeats` just doesn't work. And I don't know why.

Reply:
maybe you can add a failpoint test to see where goes wrong